torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -23,6 +24,43 @@ using torch::autograd::variable_list;
 static constexpr char* kNumBackwardPasses = "num_current_backward_passes";
 static constexpr char* kNumAutogradContexts = "num_autograd_contexts";
 
+// This hook does 2 things:
+//   1. Accumuate the gard to RPC context.
+//   2. Call post hooks of the original AccumulateGrad.
```

Comment:
Is there a reason why you call the post hooks but not the pre-hooks here?

Reply:
`autogradContext_->accumulateGrad` calls the pre-hooks, although it would be nice to move that part out into this function such that the structure looks like:

1) Call pre-hooks.
2) Accumulate Grad
3) Call post-hooks

We should also rename `AccumulateGrad::callHooks` to `AccumulateGrad::callPreHooks`

Reply:
Should we address what @pritamdamania87 suggested in another PR?

Reply:
@pritamdamania87 I don't think we should rename "callHooks". It is not calling the pre-hooks but the hooks that are associated with the leaf Variable that is given.

Reply:
I was referring to them as pre-hooks since they're a list of `FunctionPreHook`. Maybe we can rename it to `callVariableHooks` or something. @hczhu We can look at this in a separate PR as you suggested.

Reply:
> Is there a reason why you call the post hooks but not the pre-hooks here?

I didn't think about that at all, since the goal of this hook is to trigger DDP post hooks.
@pritamdamania87 @albanD How do you think? Should we call pre hooks as well here?

Reply:
Yes, Pritam mentionned it in the other comment below.